src/mqueryfront/src/QueryPage.js

```diff
@@ -162,16 +170,31 @@ class QueryPage extends Component {
             />
         );
         var queryResults = (
-            <QueryResultsStatus
-                qhash={this.state.qhash}
-                job={this.state.job}
-                matches={this.state.matches}
-            />
+            <div>
+                <button
+                    type="button"
+                    className="btn btn-primary btn-sm pull-left mr-4"
+                    onClick={this.collapsePane}
+                >
+                    <span className="fa fa-align-left" />{" "}
+                    {this.state.collapsed ? "Show" : "Hide"} query
+                </button>
+                <QueryResultsStatus
+                    qhash={this.state.qhash}
+                    job={this.state.job}
+                    matches={this.state.matches}
+                />
+            </div>
         );
         return (
             <div className="container-fluid">
                 <div className="row">
-                    <div className="col-md-6">
+                    <div
+                        id="query-column"
+                        className={
```

Comment:
See comment above, I think this ID can be replaced by class (after the CSS update:
```html
                    <div
                        className={
                            "query-column " + 
                            (this.state.collapsed ? "is-collapsed" : "col-md-5")
                        }
                    >
```